web/html/src/manager/admin/setup/products/products.tsx

```diff
@@ -1057,27 +1095,87 @@ class CheckListItem extends React.Component<CheckListItemProps> {
 }
 
 const ChannelsPopUp = (props) => {
-  const titlePopup = t("Product Channels - ") + (props.item != null ? props.item.label : "");
-  const contentPopup =
-    props.item != null ? (
-      <div>
-        <ChannelList
-          title={t("Mandatory Channels")}
-          items={props.item.channels
-            .filter((c) => !c.optional)
-            .sort((a, b) => a.label.toLowerCase().localeCompare(b.label.toLowerCase()))}
-          className={"product-channel-list"}
-        />
-        <ChannelList
-          title={t("Optional Channels")}
-          items={props.item.channels
-            .filter((c) => c.optional)
-            .sort((a, b) => a.label.toLowerCase().localeCompare(b.label.toLowerCase()))}
-          className={"product-channel-list"}
-        />
-      </div>
-    ) : null;
-  return <PopUp id="show-channels-popup" title={titlePopup} content={contentPopup} className="modal-xs" />;
+  const [checked, setChecked] = React.useState<boolean[]>([]);
+  const [installed, setInstalled] = React.useState<boolean>(props.item.status === _PRODUCT_STATUS.installed);
+
+  let mandatoryChannels;
+  let optionalChannels;
+
+  mandatoryChannels = props.item.channels
```

Comment:
Completely optional, since we already use lodash in a number of places around the project, this block could be turned around e.g.

```ts
import _partition from "lodash/partition";

// ...

const sorted = props.items.channels.sort((a, b) => a.label.toLowerCase().localeCompare(b.label.toLowerCase()));
const [mandatoryChannels, optionalChannels] = _partition(sorted, c => !c.optional);
```

That way we iterate the whole array twice, as opposed to four times. Realistically I don't think there is any difference, I just got [nerd sniped](https://xkcd.com/356/) a bit by this.

Reply:
That's actually pretty cool :smiley: 
I'll make the change TYVM